src/parsers/manifest/dash/indexes/base.ts

```diff
@@ -165,16 +165,14 @@ export default class BaseRepresentationIndex implements IRepresentationIndex {
                                     representationId,
                                     representationBitrate);
 
-    // TODO If indexRange is either undefined or behind the initialization segment
+    // TODO If indexRange is behind the initialization segment
     // the following logic will not work.
-    // However taking the nth first bytes like `dash.js` does (where n = 1500) is
-    // not straightforward as we would need to clean-up the segment after that.
-    // The following logic corresponds to 100% of tested cases, so good enough for
-    // now.
-    const range : [number, number] | undefined =
-      index.initialization !== undefined ? index.initialization.range :
-      index.indexRange !== undefined ? [0, index.indexRange[0] - 1] :
-                                       undefined;
+    let range: [number, number] | undefined;
+    if (index.initialization != null) {
+      range = index.initialization.range;
+    } else if (index.indexRange != null) {
```

Comment:
isn't it `undefined`? Why changing the previous code here?

Reply:
Done